s/transient_one_for_one_test.go

```diff
@@ -0,0 +1,180 @@
+package s_test
+
+//
+// NOTE: If you feel is counter-intuitive to have workers start before
+// supervisors in the assertions bellow, check stest/README.md
+//
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/capatazlib/go-capataz/c"
+	"github.com/capatazlib/go-capataz/s"
+	. "github.com/capatazlib/go-capataz/stest"
+)
+
+func TestTransientOneForOneSingleFailingChildRecovers(t *testing.T) {
+	parentName := "root"
+	// Fail only one time
+	child1, failChild1 := FailOnSignalChild(1, "child1", c.WithRestart(c.Transient))
+
+	events, err := ObserveSupervisor(
+		context.TODO(),
+		parentName,
+		[]s.Opt{
+			s.WithChildren(child1),
+		},
+		func(em EventManager) {
+			// NOTE: we won't stop the supervisor until the child has failed at least
+			// once
+			evIt := em.Iterator()
+			// 1) Wait till all the tree is up
+			evIt.SkipTill(SupervisorStarted("root"))
+			// 2) Start the failing behavior of child1
+			failChild1()
+			// 3) Wait till first restart
+			evIt.SkipTill(WorkerStarted("root/child1"))
+		},
+	)
+
+	assert.NoError(t, err)
+
+	AssertExactMatch(t, events,
+		[]EventP{
+			// start children from left to right
+			WorkerStarted("root/child1"),
+			SupervisorStarted("root"),
+			// ^^^ 1) failChild1 starts executing here
+			WorkerFailed("root/child1"),
+			// ^^^ 2) And then we see a new (re)start of it
+			WorkerStarted("root/child1"),
+			// ^^^ 3) After 1st (re)start we stop
```

Comment:
This is a little out of context, but the comment for `FailOnSignalChild()` indicates that after the given number of failures, it waits until the `done` from the supervisor. However, it seems the supervisor would stop when the child stops. I see the `done` channel read in `FailOnSignalChild()`. It seems circular to me. What triggers the supervisor to stop, if the function returned by `FailOnSignalChild()` is stuck waiting for `done`? Where is the child told to complete, which would in turn complete the supervisor?

Reply:
The supervisor stopping procedure is the responsibility of the [`ObserveSupervisor`](https://github.com/capatazlib/go-capataz/blob/d97d2fddf1b83cb31bb1ddbda4d675d20d6c930b/stest/assertions.go#L265) test helper. 

The supervisor stop gets executed once the callback passed to the `ObserveSupervisor` call returns.